app/code/core/Mage/Review/Block/Customer/View.php

```diff
@@ -74,10 +73,15 @@ public function getProductData()
      */
     public function getReviewData()
     {
-        if ($this->getReviewId() && !$this->getReviewCachedData()) {
-            $this->setReviewCachedData(Mage::getModel('review/review')->load($this->getReviewId()));
-        }
-        return $this->getReviewCachedData();
+        return Mage::registry('current_review');
+    }
+
+    /**
+     * @return int
+     */
+    public function getReviewId()
+    {
+        return $this->getReviewData()->getId();
```

Comment:
this will throw error when `current_review` is not set. Previously calling getReviewId() returned null 
.
What about `getData('review_id')` -> this will not work any more, right?

So the change is a breaking change, but it doesn't have to be.

Reply:
What would you suggest, setting the reviewId in the constructor?